ignite/handlers/ema_handler.py

```diff
@@ -0,0 +1,207 @@
+
+    Examples:
+          .. code-block:: python
+
+              device = torch.device("cuda:0")
+              model = nn.Linear(2, 1).to(device)
+              # update the ema every 5 iterations
+              ema_handler = EMAHandler(
+                  model, momentum_start=0.0001, momentum_end=0.0002, warmup_iters=10000)
+              # get the ema model, which is an instance of nn.Module
+              ema_model = ema_handler.ema_model
+              trainer = Engine(train_step_fn)
+              to_load = {"model": model, "ema_model", ema_model, "trainer", trainer}
+              if resume_from is not None:
+                  Checkpoint.load_objects(to_load, checkpoint=resume_from)
+
+              # update the EMA model every 5 iterations
+              ema_handler.attach(trainer, name="ema_momentum", event=Events.ITERATION_COMPLETED(every=5))
+
+              # add other handlers
+              to_save = to_load
+              ckpt_handler = Checkpoint(to_save, DiskSaver(...), ...)
+              trainer.add_event_handler(Events.EPOCH_COMPLETED, ckpt_handler)
+
+              # current momentum can be retrieved from engine.state,
+              # the attribute name is the `name` parameter used in the attach function
+              @trainer.on(Events.ITERATION_COMPLETED):
+              def print_ema_momentum(engine):
+                  print(f"current momentum: {engine.state.ema_momentum}"
+
+              # use ema model for validation
+              val_step_fn = get_val_step_fn(ema_model)
+              evaluator = Engine(val_step_fn)
+
+              @trainer.on(Events.EPOCH_COMPLETED)
+              def run_validation(engine):
+                  engine.run(val_data_loader)
+
+              trainer.run(...)
+
+          The following example shows how to attach two handlers to the same trainer:
+
+          .. code-block:: python
+
+              generator = build_generator(...)
+              discriminator = build_discriminator(...)
+
+              gen_handler = EMAHandler(generator)
+              disc_handler = EMAHandler(discriminator)
+
+              step_fn = get_step_fn(...)
+              engine = Engine(step_fn)
+
+              # update EMA model of generator every 1 iteration
+              gen_handler.attach(engine, "gen_ema_momentum", event=Events.ITERATION_COMPLETED)
+              # update EMA model of discriminator every 2 iteration
+              disc_handler.attach(engine, "dis_ema_momentum", event=Events.ITERATION_COMPLETED(every=2))
+
+              @engine.on(Events.ITERATION_COMPLETED)
+              def print_ema_momentum(engine):
+                  print(f"current momentum for generator: {engine.state.gen_ema_momentum}")
+                  print(f"current momentum for discriminator: {engine.state.disc_ema_momentum}")
+
+              engine.run(...)
+
+
+    """
+
+    def __init__(
+        self, model: nn.Module, momentum_start: float = 0.0001, momentum_end: float = 0.0002, warmup_iters: int = 100,
```

Comment:
Question, can we configure it without any warmup somehow ? It is always useful to perform a warm-up ?

Reply:
> Can we configure it without any warmup somehow?
Yes, we can set `momentum_start` = `momentum_end`, so the momentum will be constant. For the future, we can use the `ignite`'s parameter scheduling as an optional argument. So if no scheduling is provided, we keep the constant momentum.

> Is it always useful to perform a warm-up?
In the early stage, the training is not stable, so we hope to keep a small momentum. Usually the the warmup will take 3-5 epoch.  But no empirical study shows that it is ALWAYS useful.

Reply:
Thanks for details ! Honestly, I prefer the previous API https://github.com/pytorch/ignite/pull/2098/commits/58312f7f36ffd7baea40b4948f847362d9e11926 : 
```
momentum: float = 0.0002,
momentum_warmup: Optional[float] = None, 
warmup_iters: Optional[int] = None,
```

In this case, once state parameter scheduling is done, warmup args can be deprecated or removed if no release was done before...

Reply:
@sandylaker what do you think about the above comment ^^^ ?

Reply:
@vfdev-5 Do you mean only change the name `momentum_start` and `momentum_end` in commit [3bfb9c7](https://github.com/pytorch/ignite/pull/2098/commits/3bfb9c7799626db2ddd89945219aecdd57ff7c8c) back to `momentum_warmup` and `momentum`? I have submitted a new commit to handle this issue.

Reply:
Yes, thanks! Also let's set by default the following : 
```
momentum: float = 0.0002,
momentum_warmup: Optional[float] = None, 
warmup_iters: Optional[int] = None,
```
such that there is no warm-up by default.

Reply:
I updated in the most recent commit. I also added tests to test if the momentum is constant, when at least one of `momentum_warmup` and `warmup_iters` is ``None``.